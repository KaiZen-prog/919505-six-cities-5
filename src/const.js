export const RATING_SCALE_MULTIPLIER = 20;

export const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const SortingTypes = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

export const CitiesContainersClasses = {
  MAIN: {
    WITH_OFFERS: `page__main page__main--index`,
    NO_OFFERS: `page__main page__main--index page__main--index-empty`
  },

  CONTAINER: {
    WITH_OFFERS: `cities__places-container container`,
    NO_OFFERS: `cities__places-container cities__places-container--empty container`
  }
};

export const OfferCardArticleClasses = {
  MAIN_SCREEN: `cities__place-card place-card`,
  OFFER_SCREEN: `near-places__card place-card`
};

export const OfferCardImgWrapperClasses = {
  MAIN_SCREEN: `cities__image-wrapper place-card__image-wrapper`,
  OFFER_SCREEN: `near-places__image-wrapper place-card__image-wrapper`
};

export const CityMapClasses = {
  MAIN_SCREEN: `cities__map map`,
  OFFER_SCREEN: `property__map map`
};

export const CityMapSettings = {
  CITY: [52.38333, 4.9],
  ZOOM: 12,
  DEFAULT_ICON_URL: `img/pin.svg`,
  ACTIVE_ICON_URL: `img/pin-active.svg`,
  ICON_SIZE: [30, 30]
};
