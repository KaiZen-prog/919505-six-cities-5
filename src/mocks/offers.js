import {nanoid} from "nanoid";
import moment from "moment";

import {
  getRandomArrayElement,
  getRandomBool,
  getRandomInteger,
  createRandomArray
} from "../utils/common.js";

const OFFERS_QUANTITY = 4;

const TITLES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`
];

const USERS = [
  {
    avatar: `avatar-angelina.jpg`,
    name: `Angelina`
  },

  {
    avatar: `avatar-max.jpg`,
    name: `Max`
  }
];

const POSTERS = [
  `apartment-01.jpg`,
  `apartment-02.jpg`,
  `apartment-03.jpg`,
  `studio-01.jpg`,
  `room.jpg`
];

const APARTMENT_TYPES = [
  `Apartment`,
  `Private room`
];

const PRICES = {
  MIN: 10,
  MAX: 500
};

const RATING_SCALE = {
  MIN: 0,
  MAX: 50
};

const BEDROOMS = {
  MIN: 1,
  MAX: 5
};

const MAX_ADULTS = {
  MIN: 1,
  MAX: 5
};

const FEATURES = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cable TV`,
  `Fridge`,
];

const DESCRIPTION = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui. Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`;

const generateReview = () => ({
  author: getRandomArrayElement(USERS),
  rating: getRandomInteger(RATING_SCALE.MIN, RATING_SCALE.MAX) / 10,
  text: DESCRIPTION.split(`.`, getRandomInteger(1, 5)).toString(),
  date: moment(Date.now() - getRandomInteger(0, 5 * 12 * 30 * 24 * 60 * 60 * 1000)).format(`YYYY-MM-DD`),
});

const getReviews = (num) => Array.from({length: num}, generateReview);

const generateOffer = () => ({
  id: nanoid(),
  title: getRandomArrayElement(TITLES),
  poster: getRandomArrayElement(POSTERS),
  photos: createRandomArray(POSTERS),
  type: getRandomArrayElement(APARTMENT_TYPES),
  price: getRandomInteger(PRICES.MIN, PRICES.MAX),
  rating: getRandomInteger(RATING_SCALE.MIN, RATING_SCALE.MAX) / 10,
  bedroomsQuantity: getRandomInteger(BEDROOMS.MIN, BEDROOMS.MAX),
  maxAdults: getRandomInteger(MAX_ADULTS.MIN, MAX_ADULTS.MAX),
  features: createRandomArray(FEATURES),
  description: DESCRIPTION.split(`.`, getRandomInteger(1, 5)).toString(),
  isPremium: getRandomBool(),
  owner: getRandomArrayElement(USERS),
  reviews: getReviews(getRandomInteger(0, 5)),
  isInBookmarks: getRandomBool()
});

const getOffers = () => {
  const offers = [];
  for (let i = 0; i < OFFERS_QUANTITY; i++) {
    let newOffer = generateOffer();
    offers.push(newOffer);
  }

  return offers;
};

export default getOffers;
