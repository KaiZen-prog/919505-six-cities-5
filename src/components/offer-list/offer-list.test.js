import React from "react";
import renderer from "react-test-renderer";
import OfferList from "./offer-list";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {offers, InitialState} from "../../__mocks__/mocks";
import {OfferCardArticleClasses, OfferCardImgWrapperClasses, FavoriteButtonTypes} from "../../const";

const mockStore = configureStore();
const store = mockStore(InitialState);

describe(`OfferList render correctly`, () => {
  it(`OfferList on main screen`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <OfferList
                offers={offers}
                offerCardArticleClass={OfferCardArticleClasses.MAIN_SCREEN}
                offerCardImgWrapperClass={OfferCardImgWrapperClasses.MAIN_SCREEN}
                favoriteButtonType={FavoriteButtonTypes.MAIN_SCREEN}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`OfferList on favorites screen`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <OfferList
                offers={offers}
                offerCardArticleClass={OfferCardArticleClasses.FAVORITES_SCREEN}
                offerCardImgWrapperClass={OfferCardImgWrapperClasses.FAVORITES_SCREEN}
                favoriteButtonType={FavoriteButtonTypes.FAVORITES_SCREEN}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`OfferList on nearby offers`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <OfferList
                offers={offers}
                offerCardArticleClass={OfferCardArticleClasses.OFFER_SCREEN}
                offerCardImgWrapperClass={OfferCardImgWrapperClasses.OFFER_SCREEN}
                favoriteButtonType={FavoriteButtonTypes.OFFER_SCREEN}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
