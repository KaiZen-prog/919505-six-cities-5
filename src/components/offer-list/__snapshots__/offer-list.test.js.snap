// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList render correctly OfferList on favorites screen 1`] = `
Array [
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          id="card-name-link"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            218
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "92%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`OfferList render correctly OfferList on main screen 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          id="card-name-link"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            218
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "92%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`OfferList render correctly OfferList on nearby offers 1`] = `
Array [
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          id="card-name-link"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        id="card-image-link"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="\\"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg\\""
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            218
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "92%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          id="card-name-link"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;
